src/ILCompiler.Compiler/src/Compiler/NameMangler.cs

```diff
@@ -345,5 +348,40 @@ private string ComputeMangledFieldName(FieldDesc field)
 
             return mangledName;
         }
+
+        private ImmutableDictionary<string, string> _mangledStringLiterals = ImmutableDictionary<string, string>.Empty;
+
+        public string GetMangledStringName(string literal)
```

Comment:
(Or maybe just make it do what NameMangler does - prefix things with compilation unit prefixes and guarantee uniqueness by walking the entire string heap in the image and assigning numbers to things that conflict. This would also solve the SHA1 problem...)